opentelemetry-sdk/src/metrics/internal/aggregate.rs

```diff
@@ -0,0 +1,185 @@
+use std::{marker, sync::Arc};
+
+use once_cell::sync::Lazy;
+use opentelemetry::KeyValue;
+
+use crate::{
+    metrics::data::{Aggregation, Gauge, Temporality},
+    AttributeSet,
+};
+
+use super::{
+    histogram::Histogram,
+    last_value::LastValue,
+    sum::{PrecomputedSum, Sum},
+    Number,
+};
+
+const STREAM_CARDINALITY_LIMIT: u32 = 2000;
+pub(crate) static STREAM_OVERFLOW_ATTRIBUTE_SET: Lazy<AttributeSet> = Lazy::new(|| {
+    let key_values: [KeyValue; 1] = [KeyValue::new("otel.metric.overflow", "true")];
+    AttributeSet::from(&key_values[..])
+});
```

Comment:
I get that its convenient/simple to represent a set of one attribute as an `AttributeSet`, but internally, that is a `BTreeSet`, which seems a bit heavy weight to carry around one constant attribute.  So, I guess I'm wondering out loud, is a lazily created static `AttributeSet` the only way to represent this, or could a `const KeyValue` be used implementing whatever trait might be used to access/enumerate the set?  (This is a question for my own education, rather than any objection to the PR.)

Reply:
This was moved unchanged from the previous location, can certainly consider changing it but maybe in another PR as this one already has a few fiddly changes in it?